uploader.go

```diff
@@ -184,22 +192,47 @@ func (u UploadOptions) contextWithTimeout(ctx context.Context) (context.Context,
 	return context.WithTimeout(ctx, u.BatchWindow)
 }
 
+func addToPendingBatch(u UploadOptions, inputData InputData) Batch {
+	blob := inputData.GetBlob()
+	objectType := inputData.GetObjectType()
+	b, ok := pendingBatches[objectType]
+
+	// start a new batch
```

Comment:
We'll need to lock around batch creation and closure.

Reply:
It should be single-threaded, no?